core/src/main/scala/org/ensime/indexer/SearchService.scala

```diff
@@ -289,6 +283,10 @@ class SearchService(
   }
 }
 
+object SearchService {
+  type SourceInfo = (String, String, Option[String])
```

Comment:
perhaps better as a `case class` with named fields? Raw like this makes it hard to remember what the tuple entries are.